### PR TITLE
fix: skip delete if file doesn't exists

### PR DESCRIPTION
**Pull request by nickchen**
### Proposed changes

- skip delete and log if intended delete file doesn't exists
- additional changes to tests
### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [X] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

File: src/core/nginx.go [hunk 1]
```diff
@@ -399,6 +399,10 @@ func (n *NginxBinaryType) WriteConfig(config *proto.NginxConfig) (*sdk.ConfigApp
 		if _, ok = fileDeleted[file]; ok {
 			continue
 		}
+		if found, _ := FileExists(file); !found {
+			log.Debugf("skip delete for non-existing file: %s", file)
```
Comment (ornj) [inline src/core/nginx.go:403]: Might there be any information in the error returned by `FileExists` that would be potentially useful to have in the log?

Comment (nickchen) [reply src/core/nginx.go:403]: sure, will change the log

Comment (nickchen) [reply src/core/nginx.go]: added error check

**Review by oliveromahony — APPROVED** [review]

File: src/core/nginx.go [hunk 2]
```diff
@@ -399,6 +399,16 @@ func (n *NginxBinaryType) WriteConfig(config *proto.NginxConfig) (*sdk.ConfigApp
...
 		}
+
+		if found, foundErr := FileExists(file); !found {
+			if foundErr == nil {
+				log.Debugf("skip delete for non-existing file: %s", file)
+				continue
+			}
+			// possible perm deny, depends on platform
+			log.Warnf("file exists returned for %s: %s", file, foundErr)
+			return nil, foundErr
```
[the remark this answers is not in the document]
Comment (nickchen) [reply src/core/nginx.go:410]: will fix.

**Review by dhurley — APPROVED** [review]

**Pull request merged**